packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -191,7 +191,10 @@ class _BottomSheetState extends State<BottomSheet> {
     return renderBox.size.height;
   }
 
-  bool get _dismissUnderway => widget.animationController!.status == AnimationStatus.reverse;
+  late final AnimationController _effectiveAnimationController =
```

Comment:
The aniamtioncontroller created by BottomSheet.createAnimationController needs to be disposed.

Also, we need to add logic for when the widget configuration is updated (didUpdateWidget) and `widget.animationController` is no longer null.

Reply:
Not sure what to do in `didUpdateWidget` since there is no listeners registered in the controller. Do you have any tip?

Reply:
In didUpdateWidget, if animationController is no longer null, you'd have to dispose the one created from `BottomSheet.createAnimationController(this)` and use the one from widget instead. Similarly, if animationController changes to null, you'd have to instantiate a new animationcontroller in didUpdateWidget.

Reply:
Right. So let's try making controller required and non-nullable and see if it breaks any tests otherwise I will implement what you suggested above.